### DISCUSSION TITLE
DHT11 Sensor Library v2.1.0 for Arduino - Hotfix Release

### DISCUSSION THREAD

**Discussion by dhrubasaha08** [dhrubasaha08/DHT11#15] [Announcements]
# DHT11 Sensor Library v2.1.0 for Arduino - Hotfix Release

I'm thrilled to announce the release of the DHT11 Sensor Library v2.1.0 for Arduino. This version introduces several enhancements, optimizations, and a significant hotfix aimed at improving functionality and user experience. Importantly, this release is backward compatible with previous versions, ensuring a seamless transition for your projects.

## What's Changed
- **New `readTemperatureHumidity` Method**: I've introduced a method for efficiently reading both temperature and humidity simultaneously, significantly streamlining the data acquisition process.
  
- **Codebase Refinement**: I've undertaken comprehensive refactoring of the internal data reading process to reduce code repetition and enhance maintainability.

- **Advanced Error Handling**: I've improved the error handling mechanism. Now, specific error codes are returned by the `readTemperatureHumidity` method, aligning it with the `readTemperature` and `readHumidity` methods for consistency.

- **Documentation Enhancements**: I've updated the documentation to reflect the latest changes, providing clear guidance on utilizing the new library version effectively.

### Acknowledgements
A special thanks to the community members who have contributed to this release through their valuable feedback and suggestions.

## Getting Started
To take advantage of the new features and improvements, please update your library to the latest version. You can find detailed instructions on installation and usage in our [documentation](https://github.com/dhrubasaha08/DHT11).

## Feedback and Contributions
Your feedback is crucial to the continuous improvement of the DHT11 Sensor Library. If you encounter any issues or have suggestions for future releases, please feel free to open an issue or submit a pull request.

Thank you for your continued support and contributions to the DHT11 Sensor Library project. I'm excited to see the innovative projects you'll create with these new enhancements!

<hr /><em>This discussion was created from the release <a href='https://github.com/dhrubasaha08/DHT11/releases/tag/v2.1.0'>DHT11 Sensor Library v2.1.0 for Arduino - Hotfix Release</a>.</em>

**Comment by cliffom** [comment]
Awesome work! Updated my project to use the update version and I'm not experiencing any of the timeout issues I had with 2.0.0. Thank you for this and the new functionality!
